File: Linear/Static/Array/05_Searching/Linear Search/LinearSearch.c
#include <stdio.h>

int linearSearch(int arr[], int n, int x) {
    // Iterate over the array in order to
    // find the key x
    for (int i = 0; i < n; i++) {
        if (arr[i] == x)
        return i;
        
    }
    
    return -1;
}

int main() {
    int arr[] = {10, 8, 2, 7, 3, 4, 9, 1, 6, 5};
    int n = sizeof(arr) / sizeof(arr[0]);
    int x;

    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    
    printf("\nEnter a number to search: ");
    scanf("%d", &x);

    int result = linearSearch(arr, n, x);
    (result == -1) ? printf("%d is not present in Array", x) : printf("%d is present at index %d", x, result);

    return 0;
}

File: Linear/Static/Array/05_Searching/Linear Search/LinearSearch.py
def linearSearch(arr, x):
    # Iterate over the array in order to
    # find the key x
    for i in range(len(arr)):
        if arr[i] == x:
            return i
    return -1

arr = [10, 8, 2, 7, 3, 4, 9, 1, 6, 5]

for i in range(len(arr)):
    print(arr[i], end=" ")

x = int(input("\nEnter a number to search: "))

result = linearSearch(arr, x)
if result == -1:
    print(x, "is not present in Array")
else:
    print(x, "is present at index", result)